metadata-service/auth-ranger-impl/src/main/java/com/datahub/authorization/ranger/DataHubRangerClientImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.datahub.authorization.ranger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.util.StringUtils;
+import org.apache.ranger.RangerClient;
+import org.apache.ranger.RangerServiceException;
+import org.apache.ranger.plugin.audit.RangerDefaultAuditHandler;
+import org.apache.ranger.plugin.policyengine.RangerAccessRequest;
+import org.apache.ranger.plugin.policyengine.RangerAccessResult;
+import org.apache.ranger.plugin.service.RangerBasePlugin;
+
+
+@Slf4j
+public class DataHubRangerClientImpl implements DataHubRangerClient {
+  public static final String SERVICE_NAME = "datahub";
+  public static final String PROP_RANGER_URL = StringUtils.format("ranger.plugin.%s.policy.rest.url", SERVICE_NAME);
+  private static RangerBasePlugin rangerBasePlugin;
+  private final AuthorizerConfig authorizerConfig;
+  private RangerClient rangerClient;
+
+  public DataHubRangerClientImpl(AuthorizerConfig authorizerConfig) {
+    this.authorizerConfig = authorizerConfig;
+  }
+
+  @Override
+  public void init() {
+
+    // Don't need any synchronization as single Authorizer object is getting created on server startup
+    if (rangerBasePlugin == null) {
+      // rangerBasePlugin is static
+      // Make sure classpath should have ranger-datahub-security.xml file
+      rangerBasePlugin = this.newRangerBasePlugin();
+      rangerBasePlugin.setResultProcessor(new RangerDefaultAuditHandler());
+
+      rangerBasePlugin.init();
+      log.info("Ranger policy engine is initialized");
+    }
+
+    rangerClient = this.newRangerClient();
+  }
+
+  public RangerBasePlugin newRangerBasePlugin() {
+    return new RangerBasePlugin(SERVICE_NAME, SERVICE_NAME);
+  }
+
+  public RangerClient newRangerClient() {
+    String rangerURL = rangerBasePlugin.getConfig().get(PROP_RANGER_URL);
```

Comment:
rangerBasePlugin.init should take care of this, during the init process rangerBasePlugin is connecting to Apache Ranger to pull the policies. The above method is getting called after rangerBasePlugin.init and we are not reading those configuration files so it may add duplicate code and validation.